### ISSUE TITLE
`FetchContentsOfFilesFromAsg` assumes all instances have public IP addresses

### ISSUE THREAD

**Issue by wjam** [gruntwork-io/terratest#251]
The `FetchContentsOfFilesFromAsg` method (and underlying `FetchContentsOfFilesFromAsgE` & `FetchContentsOfFilesFromInstanceE`) assume that all instances will have a public IP address rather than be accessible via the open internet. The method should at least be accessible via private IP addresses as well as public, potentially also via a bastion box.

It may need a new method(s) to take something similar to `SshConnectionOptions`?

**Comment by brikis98** [comment]
Yes, I believe you're right. We typically try to run our automated tests in public subnets to make testing easier, but there are definitely use cases where that isn't possible/practical. The Terratest SSH code supports connection via a jump host, so a PR to update the various flavors of the  `FetchContentsOfFilesFromAsg` methods to work with jump hosts too is welcome!

**Comment by james03160927** [comment]
inactive for the past 6 years. Closing this issue as there's no more interest.  Feel free to reopen this if we still need this.
